optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-client/src/main/java/org/optaplanner/workbench/screens/solver/client/editor/TerminationPopup.java

```diff
@@ -1,168 +0,0 @@
-            case DAYS_SPENT_LIMIT:
-            case UNIMPROVED_MILLISECONDS_SPENT_LIMIT:
-            case UNIMPROVED_SECONDS_SPENT_LIMIT:
-            case UNIMPROVED_MINUTES_SPENT_LIMIT:
-            case UNIMPROVED_HOURS_SPENT_LIMIT:
-            case UNIMPROVED_DAYS_SPENT_LIMIT: {
-                view.showTerminationValueLabel( true );
-                view.showNumericTerminationValueInput( true );
-                view.showTextTerminationValueInput( false );
-                view.showListTerminationValueInput( false );
-                view.setNumericTerminationValueMaximum( Long.MAX_VALUE );
-                break;
-            }
-            case SCORE_CALCULATION_COUNT_LIMIT:
-            case STEP_COUNT_LIMIT:
-            case UNIMPROVED_STEP_COUNT_LIMIT: {
-                view.showTerminationValueLabel( true );
-                view.showNumericTerminationValueInput( true );
-                view.showTextTerminationValueInput( false );
-                view.showListTerminationValueInput( false );
-                view.setNumericTerminationValueMaximum( Integer.MAX_VALUE );
-                break;
-            }
-            case TERMINATION_COMPOSITION_STYLE: {
-                List<Pair<String, String>> terminationValueList = new ArrayList<>();
-                terminationValueList.add( new Pair<>( SolverEditorConstants.INSTANCE.And(), TerminationCompositionStyleModel.AND.name() ) );
-                terminationValueList.add( new Pair<>( SolverEditorConstants.INSTANCE.Or(), TerminationCompositionStyleModel.OR.name() ) );
-                view.initTerminationValueList( terminationValueList );
-                view.showTerminationValueLabel( true );
-                view.showNumericTerminationValueInput( false );
-                view.showTextTerminationValueInput( false );
-                view.showListTerminationValueInput( true );
-                break;
-            }
-            case BEST_SCORE_FEASIBLE: {
-                List<Pair<String, String>> terminationValueList = new ArrayList<>();
-                terminationValueList.add( new Pair<>( SolverEditorConstants.INSTANCE.True(), Boolean.TRUE.toString() ) );
-                terminationValueList.add( new Pair<>( SolverEditorConstants.INSTANCE.False(), Boolean.FALSE.toString() ) );
-                view.initTerminationValueList( terminationValueList );
-                view.showTerminationValueLabel( true );
-                view.showNumericTerminationValueInput( false );
-                view.showTextTerminationValueInput( false );
-                view.showListTerminationValueInput( true );
-                break;
-            }
-            case BEST_SCORE_LIMIT: {
-                view.showTerminationValueLabel( true );
-                view.showNumericTerminationValueInput( false );
-                view.showTextTerminationValueInput( true );
-                view.showListTerminationValueInput( false );
-                break;
-            }
```

Comment:
Add a default case that fails fast:


            default:
                throw new IllegalStateException("The terminationConfigOption ("
                        + terminationConfigOption + ") is not implemented.");

That way, if another developer (or your future self) adds a new TerminationConfigOption in a year from now and they forget to deal with it here, the issue will stick out like a sore thumb for him/her, QA, community users, etc and it will be noticed faster, give us a clear error report and it will be fixed faster. In it's current form, the user might see "nothing happening" and believe the app is stuck, slow or something and he/she might ignore it instead of reporting it.

This fail fast philosophy applies to all these kinds of switch statements.